portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -104,4 +139,14 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  private GoogleIdToken validate(String idTokenString, JacksonFactory jsonFactory, NetHttpTransport transport) throws Exception {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(transport, jsonFactory)
+      .setAudience(Collections.singletonList(CLIENT_ID))
+      .build();
```

Comment:
Creating GoogleIdTokenVerifier every time you handle a request seems suboptimal. Instead, make GoogleIdTokenVerifier an instance variable.

Reply:
Solved! Thanks for pointing it out.